emission/analysis/userinput/matcher.py

```diff
@@ -38,57 +39,132 @@ def match_incoming_inputs(user_id, timerange):
             elif ui.metadata.key in multi_non_deleted_key_list:
-                handle_multi_non_deleted_match(confirmed_trip, ui)
+                handle_multi_non_deleted_match(confirmed_obj, ui)
             else:
                 assert False, "Found weird key {ui.metadata.key} that was not in the search list"
             import emission.storage.timeseries.builtin_timeseries as estbt
-            estbt.BuiltinTimeSeries.update(confirmed_trip)
+            estbt.BuiltinTimeSeries.update(confirmed_obj)
         else:
             logging.warn("No match found for single user input %s, moving forward anyway" % ui)
         lastInputProcessed = ui
 
     return lastInputProcessed
 
-def handle_single_most_recent_match(confirmed_trip, ui):
+def handle_single_most_recent_match(confirmed_obj, ui):
     input_name = obj_to_dict_key(ui.metadata.key)
-    if input_name == "trip_user_input":
-        confirmed_trip["data"]["user_input"][input_name] = ui
+    if input_name == "trip_user_input" or input_name == "place_user_input":
+        confirmed_obj["data"]["user_input"][input_name] = ui
     else:
-        confirmed_trip["data"]["user_input"][input_name] = ui.data.label
-
-def handle_multi_non_deleted_match(confirmed_trip, ui):
-    logging.debug(f"handling user input {ui} for {confirmed_trip}")
-    if "trip_addition" not in confirmed_trip["data"] or \
-        confirmed_trip["data"]["trip_addition"] is None:
-        confirmed_trip["data"]["trip_addition"] = []
-    if "status" not in ui.data or ui.data.status == ecwtui.InputStatus.ACTIVE:
-        confirmed_trip["data"]["trip_addition"].append(ui)
-    elif ui.data.status == ecwtui.InputStatus.DELETED:
-        after_del_list = [ta for ta in confirmed_trip["data"]["trip_addition"] if ta["data"]["match_id"] != ui["data"]["match_id"]]
-        confirmed_trip["data"]["trip_addition"] = after_del_list
+        confirmed_obj["data"]["user_input"][input_name] = ui.data.label
+
+def handle_multi_non_deleted_match(confirmed_obj, ui):
+    logging.debug(f"handling user input {ui} for {confirmed_obj}")
+    if "additions" not in confirmed_obj["data"] or \
+        confirmed_obj["data"]["additions"] is None:
+        confirmed_obj["data"]["additions"] = []
+    if "status" not in ui.data or ui.data.status == ecwui.InputStatus.ACTIVE:
+        confirmed_obj["data"]["additions"].append(ui)
+    elif ui.data.status == ecwui.InputStatus.DELETED:
+        after_del_list = [ta for ta in confirmed_obj["data"]["additions"] if ta["data"]["match_id"] != ui["data"]["match_id"]]
+        confirmed_obj["data"]["additions"] = after_del_list
     else:
         # TODO: Decide whether to error or to warn here
         logging.warn("Invalid status found in user input %s, moving forward anyway" % ui)
 
+def create_composite_objects(user_id):
+    time_query = epq.get_time_range_for_composite_object_creation(user_id)
+    ts = esta.TimeSeries.get_time_series(user_id)
```

Comment:
high level cleanup: I am not sure that this function belongs here. Composite trip is not specifically related to input matching. This seems like the `OUTPUT_GEN` stage to me, so I would put this into `emission/analysis/result` or `emission/analysis/plotting`

Reply:
Agreed, it's out of place. I just didn't know where to put it 😅

`emission/analysis/plotting` looks appropriate since composite trip represents what is being presented to the user